spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -76,16 +85,20 @@ private String removePrefix(String bindingTarget) {
 	}
 
 	public void unbindConsumers(String inputChannelName) {
+		String transport = this.channelBindingServiceProperties.getBinder(inputChannelName);
+		Binder<MessageChannel> binder = binderFactory.getBinder(transport);
 		if (BinderUtils.isChannelPubSub(this.channelBindingServiceProperties.getBindingDestination(inputChannelName))) {
-			this.binder.unbindPubSubConsumers(inputChannelName, consumerGroup(inputChannelName));
+			binder.unbindPubSubConsumers(inputChannelName, consumerGroup(inputChannelName));
 		}
 		else {
-			this.binder.unbindConsumers(inputChannelName);
+			binder.unbindConsumers(inputChannelName);
 		}
 	}
 
 	public void unbindProducers(String outputChannelName) {
-		this.binder.unbindProducers(outputChannelName);
+		String transport = this.channelBindingServiceProperties.getBinder(outputChannelName);
```

Comment:
we could factor out a private `getBinderForChannel()` method given how many times these 2 lines are duplicated:

```
String transport = this.channelBindingServiceProperties.getBinder(channelName);
Binder<MessageChannel> binder = binderFactory.getBinder(transport);
```